File: AndroidStudioProjects/FCMpushnotice801/app/src/main/java/com/example/fcmpushnotice801/MainActivity.java
package com.example.fcmpushnotice801;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.util.Log;
import android.widget.TextView;

import com.google.android.gms.tasks.OnCompleteListener;
import com.google.android.gms.tasks.Task;
import com.google.firebase.messaging.FirebaseMessaging;

public class MainActivity extends AppCompatActivity {
    public static final String TAG = "MainActivity";
    private TextView resultTextView(){
        return findViewById(R.id.resultTextView);
    }
    private TextView firebaseToken(){
        return findViewById(R.id.firebaseTokenText);
    }
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        initFirebase();
    }

    //토큰 값을 얻어옴
    public void initFirebase(){
        FirebaseMessaging.getInstance().getToken()
                .addOnCompleteListener(new OnCompleteListener<String>() {
            @Override
            public void onComplete(@NonNull Task<String> task) {
                if (task.isSuccessful()) {
                    // Get new FCM registration token
                    String token = task.getResult();
                    firebaseToken().setText(token);
                }
            }
        });
    }
}

File: AndroidStudioProjects/KTFCMpushnoticejh801/app/src/main/java/com/example/ktfcmpushnoticejh801/MainActivity.kt
package com.example.ktfcmpushnoticejh801

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.TextView
import com.google.firebase.messaging.FirebaseMessaging

class MainActivity : AppCompatActivity() {
    //게으른 초기화
    private val resultTextView: TextView by lazy{
        findViewById(R.id.resultTextView)
    }
    private val firebaseToken: TextView by lazy{
        findViewById(R.id.firebaseTokenText)
    }
    //기존의 자바방식

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
    }
    private fun initFirebase() {
        FirebaseMessaging.getInstance().token
            .addOnCompleteListener{task ->
                if(task.isSuccessful){
                    firebaseToken.text = task.result
                }
            }
    }//코틀린에서는 ?/?가 하나면 람다식으로 처리한다
}